File: src/main/java/org/moinex/model/enums/AssetType.java
/*
 * Filename: AssetType.java
 * Created on: January  2, 2026
 * Author: Lucas Araújo <araujolucas@dcc.ufmg.br>
 */

package org.moinex.model.enums;

/**
 * ENUM that represents the type of asset for allocation purposes
 * Includes both ticker types and other investment types like bonds
 */
public enum AssetType {
    STOCK,
    FUND,
    CRYPTOCURRENCY,
    REIT,
    ETF,
    BOND
}

File: src/main/kotlin/org/moinex/model/enums/AssetType.kt
/*
 * Filename: AssetType.kt (original filename: AssetType.java)
 * Created on: January  2, 2026
 * Author: Lucas Araújo <araujolucas@dcc.ufmg.br>
 *
 * Migrate to Kotlin on 28/02/2026
 */

package org.moinex.model.enums

enum class AssetType(
    val category: AssetCategory,
) {
    STOCK(AssetCategory.TICKER),
    FUND(AssetCategory.TICKER),
    CRYPTOCURRENCY(AssetCategory.TICKER),
    REIT(AssetCategory.TICKER),
    ETF(AssetCategory.TICKER),
    BOND(AssetCategory.FIXED_INCOME),
}

enum class AssetCategory {
    TICKER,
    FIXED_INCOME,
}
